Gems/ROS2/Code/Include/ROS2/Manipulator/JointPublisherComponent.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <ROS2/Manipulator/MotorizedJointComponent.h>
+#include <RobotImporter/URDFMetadataComponent.h>
+#include <rclcpp/publisher.hpp>
+#include <sensor_msgs/msg/joint_state.hpp>
+#include <sensor_msgs/msg/joint_state.hpp>
+#include <control_msgs/action/follow_joint_trajectory.h>
+#include <AzCore/Component/TickBus.h>
+
+namespace ROS2
+{
+    //! A component responsible for storing the jointComponent tree structure
+    //! and each of the joint's name as they are described in URDF
+    class JointPublisherComponent
+        : public URDFMetadataComponent
+        , public AZ::TickBus::Handler
+    {
+    public:
+        AZ_COMPONENT(JointPublisherComponent, "{a679c2e4-a602-46de-8db4-4b33d83317f4}", AZ::Component);
+        JointPublisherComponent() = default;
+
+        void Activate() override;
+        void Deactivate() override;
+        void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+
+        static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+        static void Reflect(AZ::ReflectContext* context);
+
+    private:
+        void PublishMessage();
+        void UpdateMessage();
+        void InitializeMap();
+
+        double GetJointPosition(const AZ::Component* hingeComponent) const;
+        void InitializeJointStateMessage();
+
+        std::shared_ptr<rclcpp::Publisher<sensor_msgs::msg::JointState>> m_jointstatePublisher;
```

Comment:
Should this (and other occurrences of shared_ptr) use AZStd instead of std::?

Reply:
the other publishers share the same format (e.g. https://github.com/o3de/o3de-extras/blob/development/Gems/ROS2/Code/Source/GNSS/ROS2GNSSSensorComponent.h#L47 )